cirq/protocols/apply_mixture_protocol_test.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def make_buffers(shape, dtype):
+    return (np.empty(shape, dtype=dtype), np.empty(shape, dtype=dtype),
+            np.empty(shape, dtype=dtype))
+
+
+def apply_mixture(val,
```

Comment:
This needs a better name, and type annotations. Maybe it should be refactored into a "applying X to Y produces result Z" testing method?

Reply:
This also happens in the density matrix channel apply tests as well I'm totally up for doing it but I think it should be in a separate PR.